### ISSUE TITLE
prefer-const: incorrectly detects modified variable as unmodified

### ISSUE THREAD

**Issue by logotype** [eslint/eslint#4570]
The version of ESLint you are using: `1.4.0`

Source code that caused the problem:
```javascript
let md5blks = [],
    i = 0;
for (i; i < 64; i += 4) {
    md5blks[i >> 2] = ‘some value’;
}
```

The configuration you're using:
`"prefer-const": 2`

The actual ESLint output complete with line numbers:
```javascript
Message:
    `md5blks` is never modified, use `const` instead.
Details:
    fileName: src/hash/MD5.js
    lineNumber: 149

  149:13  error    `md5blks` is never modified, use `const` instead  prefer-const
```

Complete source here: 
https://github.com/logotype/hashdog/blob/47738271a6f20122c263f392d4e9f0ab407ad300/src/hash/MD5.js#L148

**Comment by platinumazure** [comment]
The const keyword refers to variable references being unchanged, not that
properties are not modifiable. This is working as designed.

**Comment by gaurav21r** [comment]
For anyone else landing up here, you might find this helpful :point_down: 

http://stackoverflow.com/questions/23436437/why-can-i-change-value-of-a-constant-in-javascript
